frontend/src/utils/account-send.ts

```diff
@@ -0,0 +1,89 @@
+import { encrypt, decrypt } from 'ethereum-cryptography/aes';
+import { hexToBytes, utf8ToBytes, bytesToUtf8 } from 'ethereum-cryptography/utils';
+import { getRandomBytesSync } from 'ethereum-cryptography/random';
+import localforage from 'localforage';
+import { BigNumber } from './ethers';
+
+const LOCALFORAGE_KEY = 'acountSends';
+
+type SendMetadataEncrypted = {
+  encryptedAddress: Uint8Array;
+  amount: string;
+  tokenAddress: string;
+  dateSent: Date;
+  hash: string;
+  iv: Uint8Array;
+};
+
+type SendMetadataDecrypted = {
+  amount: string;
+  address: string;
+  dateSent: Date;
+  hash: string;
+  tokenAddress: string;
+};
+
+const encryptAddress = async (recipientAddress: string, privateKey: string) => {
+  const iv = getRandomBytesSync(16);
+  const encrypted = await encrypt(utf8ToBytes(recipientAddress), hexToBytes(privateKey), iv, 'aes-256-cbc');
```

Comment:
From a conversation with @seresistvanandras, we should be able to encrypt with xor as follows:
- Start encryption counter `c = 0`
- On first send, encryption key `k` is `keccak256(viewingPrivateKey, c)`
- Take 160 bit recipient address, append one bit indicating whether or not advanced mode was enabled, and a second bit indicated whether the checkbox was checked, giving 162 bits. Call this data `d`
- Pad `d` with zeros so it's length is 256 bits. Padding direction probably doesn't matter, so let's right pad since that's how `bytesNN` behaves in solidity and I think ethers.js has methods to do that (hexZeroPad)
- Encrypted address `a = k ^ d` where `^` is xor
- Increment `c` to `1`, save `a` and `c` in browser storage (we of course only need to store the new (current) value of `c` since it increments, but should save all `a` values)
- Decrypt with `d = a ^ k` 

This should be safe because:
- With AES or this approach, compromised viewing private key means compromised send history
- Even if you have `k`, you can't back out the viewing key from it even if you know the counter values, thanks to the hashing

Nice benefits of the xor + counter approach:
- Supports random access into the history at any point, i.e. you can decrypt just the 10th tx sent. This doesn't really matter currently but may be helpful one day for people with hundreds/thousands of sends
- If local storage is cleared, and you know with certainty that tx `i` was sent to recipient `r`, you can figure out the required counter value to decrypt that send correctly and then decrypt the rest from there
- No new encryption deps needed

Reply:
> encrypted address `a`, is also written to the blockchain or not? Or is it only stored in the browser of the user?

`a` would only be stored locally in the user's browser.

The rest of this makes sense to me though, thanks for confirming things!